src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -103,6 +108,16 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
                             linkOptions: DataflowOption.PropagateCompletion,
                             cancellationToken: cancellationToken),
 
+                    ProjectDataSources.SyncLinkTo(
+                        _activeConfiguredProjectProvider.ActiveConfiguredProjectBlock.SyncLinkOptions(),
+                        _projectBuildSnapshotService.SourceBlock.SyncLinkOptions(),
+                        target: DataflowBlockFactory.CreateActionBlock<IProjectVersionedValue<ValueTuple<ConfiguredProject, IProjectBuildSnapshot>>>(e =>
+                                ApplyBuildSnapshotChanged(e),
+                            _project.UnconfiguredProject,
+                            ProjectFaultSeverity.LimitedFunctionality),
+                        linkOptions: DataflowOption.PropagateCompletion,
+                        cancellationToken: cancellationToken),
```

Comment:
Instead of creating a new link here, add the project build snapshot into the above build link.

The new data source must be sync-linked with other sources to avoid races.

`OnProjectChangedAsync` is called for both evaluation and build results. You will likely need to add another member to the `ValueTuple` for the new data, and have it as `null` for evaluation data. Then you can pass the `IProjectBuildSnapshot` through to `ApplyProjectBuildAsync`.